src/app/core/common-components/entity-select/entity-select.component.ts

```diff
@@ -297,10 +312,22 @@ export class EntitySelectComponent<E extends Entity> implements OnChanges {
   }
 
   private emitChange() {
-    this.selectionChange.emit(this.selectedEntities.map((e) => e.getId()));
+    const newSelection = this.selectedEntities.map((e) => e.getId());
+
+    if (isMulti(this)) {
+      this.selectionChange.emit(newSelection);
+    } else {
+      this.selectionChange.emit(newSelection[0] as T);
```

Comment:
A bit annoying that the correct type `T = string` is not picked up for the `false` case here.